### ISSUE TITLE
Error with Firewall Data Casing 

### ISSUE THREAD

**Issue by blueandgold** [forseti-security/forseti-security#3325]
Testing 2.23.0 release, and running the firewall scanner and IAP scanner.

The problem is in the casing of `IPProtocol` vs `ipProtocol`.  (see last line)

```
{
  "serverMessage": "Error running scanner: IapScanner: 'Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/scanner/scanner.py\", line 119, in run\n    scanner.run()\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/scanner/scanners/iap_scanner.py\", line 682, in run\n    all_violations, _ = self._find_violations(iap_data)\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/scanner/scanners/iap_scanner.py\", line 667, in _find_violations\n    for (iap_resources, project_resource_counts) in iap_data:\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/scanner/scanners/iap_scanner.py\", line 632, in _retrieve\n
    firewall_rules = self._get_firewall_rules(parent.type_name)\n
  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/scanner/scanners/iap_scanner.py\", line 531, in _get_firewall_rules\n
    json_string=firewall_rule.data))\n
  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/common/gcp_type/firewall_rule.py\", line 221, in from_json\n
    json_dict, project_id=project_id, validate=True)\n
  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/common/gcp_type/firewall_rule.py\", line 202, in _transform\n
    return FirewallRule(validate=validate, **in_dict)\n
  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/common/gcp_type/firewall_rule.py\", line 105, in __init__\n
    self.validate()\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/common/gcp_type/firewall_rule.py\", line 293, in validate\n
    self.firewall_action.validate()\n  File \"/usr/local/lib/python3.6/dist-packages/forseti_security-2.23.0-py3.6.egg/google/cloud/forseti/common/gcp_type/firewall_rule.py\", line 698, in validate\n
    invalid_keys)\n
google.cloud.forseti.common.gcp_type.firewall_rule.InvalidFirewallActionError: Action can only have \"IPProtocol\" and \"ports\": {'ipProtocol'}\n'"
```

**Comment by blueandgold** [comment]
Looks like this is redundant data provided by CAI, for backward compatibility.
```
"allowed": [{"IPProtocol": "tcp", "ipProtocol": "tcp", "port": ["22"], "ports": ["22"]}], 
```

**Comment by blueandgold** [comment]
Forseti used to convert the casing, but it was removed in this PR.
https://github.com/forseti-security/forseti-security/pull/3276/files

**Comment by gkowalski-google** [comment]
Fixed in 2.23.0 release.
